core/win32/os.c

```diff
@@ -5759,6 +5759,23 @@ mark_page_as_guard(byte *pc)
     ASSERT(res);
 }
 
+/* Removes guard protection from page containing pc */
+bool
+unmark_page_as_guard(byte *pc, uint prot)
+{
+    uint old_prot;
+    int res;
+    byte *start_page = (byte *)ALIGN_BACKWARD(pc, PAGE_SIZE);
+
+    uint flags = memprot_to_osprot(prot & ~MEMPROT_GUARD);
+    res = protect_virtual_memory(start_page, PAGE_SIZE, flags, &old_prot);
+    /* it is likely that another thread accessed the guarded page
```

Comment:
You mean s/likely/possible/, right?  It's not expected to happen over half the time so why "likely"?

Why is this comment before the ASSERT?  Will res be false on a race?  If so, this ASSERT cannot remain as it is: we cannot have an ASSERT that will fire with legitimate app behavior.  If you want to count how often this happens or sthg, use a STATS_INC, or for a soft notify use SYSLOG_INTERNAL_WARNING or sthg.

Also note the preference for capitalization + punctuation